File: gluecodium/src/test/resources/smoke/structs/output/cpp/src/smoke/StructsWithConstants.cpp
// -------------------------------------------------------------------------------------------------
//
//
// -------------------------------------------------------------------------------------------------
#include "smoke/StructsWithConstants.h"
#include <utility>
namespace smoke {
const ::smoke::RouteUtils::RouteType StructsWithConstants::Route::DEFAULT_TYPE = ::smoke::RouteUtils::RouteType::EQUESTRIAN;
StructsWithConstants::Route::Route( )
    : description{ }, type{ }
{
}
StructsWithConstants::Route::Route( ::std::string description, ::smoke::RouteUtils::RouteType type )
    : description( std::move( description ) ), type( std::move( type ) )
{
}
}

File: gluecodium/src/test/resources/smoke/structs/output/android-kotlin/com/example/smoke/StructsWithConstants.kt
/*

 *
 */

@file:JvmName("StructsWithConstants")

package com.example.smoke


class StructsWithConstants {

    class Route {
        @JvmField var description: String
        @JvmField var type: RouteUtils.RouteType



        constructor(description: String, type: RouteUtils.RouteType) {
            this.description = description
            this.type = type
        }





        companion object {
            @JvmField final val DEFAULT_DESCRIPTION: String = "Nonsense"
            @JvmField final val DEFAULT_TYPE: RouteUtils.RouteType = RouteUtils.RouteType.EQUESTRIAN
        }
    }








}

